### DISCUSSION TITLE
How to generate a new class with Rector?

### DISCUSSION THREAD

**Discussion by anyt** [rectorphp/rector#7965] [Q&A] [answered]
Hi,
In a rector rule I need to generate a new class and place it to the new file accoring to psr-4. Can't find similar examples, maybe someone has an idea where to start.

To be more clear, the task is to generate multiple classes in different namespaces based on another single class.

Or maybe there is a way to move a class to different namespace with Rector

Thank you.

**Comment by samsonasik** [comment, marked as answer]
You may looking for `BetterStandardPrinter::prettyPrintFile()`:

https://github.com/rectorphp/rector/blob/01557664e669f840ff6bdce1e95e136c17466627/src/PhpParser/Printer/BetterStandardPrinter.php#L129

You can use native file_put_contents to fill the file with the contents.
